lib/src/models/timer.dart

```diff
@@ -0,0 +1,82 @@
+import 'package:timezone/data/latest.dart' as tz;
+import 'package:timezone/timezone.dart' as tz;
+
+import '../util/notification_service.dart';
+
+class TimerList {
+  static final TimerList _timerList =
+  TimerList._internal();
+  List<Timer> timers = <Timer>[];
+
+  factory TimerList() {
+    return _timerList;
+  }
+
+  TimerList._internal();
+
+  clear() {
+    this.timers.clear();
+    NotificationService().cancelAll();
+  }
+}
+
+class Timer {
+  final String title;
+  final String body;
+  final Duration duration;
+  int id;
+  tz.TZDateTime done;
+  final int recipeId;
+
+  Timer(this.recipeId, this.title, this.body, this.duration, [tz.TZDateTime done]) {
+    tz.initializeTimeZones();
+    this.done = tz.TZDateTime.now(tz.local).add(this.duration);
+    TimerList().timers.add(this);
```

Comment:
I see the idea of "adding" the timer itself to the timerList. But i think this "abusing" the Constructor of a data object. 

In my opinion the constructor of a model should just initialize the data object (like calculating the done field).
But the functionality that adds the timer to the timerList should be done somewhere else. 

I want to state here that this is only personal preference!

Reply:
I think you are right. This abuses the constructor. I will change it.